src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -1566,6 +1681,183 @@ def copy_external_import(self, imported_var, tag=None):
+                # otherwise ignore this step.
+                if isinstance(imported_sym, type(local_sym)):
+                    # pylint: disable=unidiomatic-typecheck
+                    if type(imported_sym) is not type(local_sym):
+                        if isinstance(imported_sym, TypedSymbol):
+                            # All TypedSymbols have a mandatory datatype
+                            # argument
+                            local_sym.specialise(
+                                type(imported_sym),
+                                datatype=imported_sym.datatype)
+                        else:
+                            local_sym.specialise(type(imported_sym))
+
+                    local_sym.copy_properties(imported_sym)
+                    # Restore the interface and visibility as these are
+                    # local (not imported) properties
+                    local_sym.interface = interface
+                    local_sym.visibility = visibility
+            else:
+                # This table did not already contain a symbol with this
+                # name.
+                if csymbol.wildcard_import:
+                    # This symbol is PUBLIC and inside a wildcard import,
+                    # so it needs to be declared in the symbol table.
+                    local_sym = imported_sym.copy()
+                    local_sym.interface = ImportInterface(csymbol)
+                    local_sym.visibility = self.default_visibility
+                    self.add(local_sym)
+            imported_symbols.add(local_sym)
+        return imported_symbols
+
+    def _update_with_resolved_symbol(self, symbol: Symbol):
+        '''
+        Given a newly-resolved symbol, walk down through the scopes below
+        the scope associated with this table and replace any instances of
+        a symbols that are now known to be this symbol.
+
+        :param symbol: the Symbol that has been resolved.
+
+        '''
+        norm_name = self._normalize(symbol.name)
+        # Import here to avoid circular dependencies
+        # pylint: disable=import-outside-toplevel
+        from psyclone.psyir.nodes import ScopingNode
+        # Walk down through the scopes below this one.
+        for scoping_node in self.node.walk(ScopingNode):
+            if scoping_node is self.node:
+                # Skip ourself.
+                continue
+            symbol_table = scoping_node.symbol_table
+            test_symbol = symbol_table.lookup(norm_name,
+                                              scope_limit=scoping_node,
+                                              otherwise=None)
+            if not test_symbol or not test_symbol.is_unresolved:
+                # Either this table doesn't contain a symbol with the
+                # same name as the imported one or it does but it is
+                # resolved (and thus shadows this one) so we ignore it.
+                continue
```

Comment:
Codecov complains that this is not tested

Reply:
It's covered for me now - we'll see what Codecov says.